### ISSUE TITLE
Problem with streaming RTCM with str2str to rtk2go.com

### ISSUE THREAD

**Issue by baxter86811** [tomojitakasu/RTKLIB#541]
Hello,

I'm having a problem with sending RTCM messages to rtk2go.com and getting a Float connection on my rover.

I have a Ardusimple simpleRTKB board with Ublox Zed-F9P that is connected with USB cable to a Raspberry Pi. Pi should send correction signal to rtk2go.com with str2str:
![107680340_1309031819488979_7537222697456746602_n](https://user-images.githubusercontent.com/68125356/87187940-455dc680-c2ee-11ea-9f56-ab0c81e35964.png)
I configured Zed-F9P to send this messages: 1005(5),1077(1),1087(1),1097(1),1230(1)

On the SNIP monitor everything looks OK:
![image](https://user-images.githubusercontent.com/68125356/87188166-a2597c80-c2ee-11ea-8252-d8c159a341eb.png)

Only thing is I can't see 1230 message on the Mount Point Summary and I can't get 3D/DGNSS/FLOAT signal on my rover (also Ardusimple simpleRTKB). I tried to use a different MountPt that is nearby and I got FLOAT signal, so I guess my rover configuration is OK. Can someone please help me out?

**Comment by DavidKelleySCSC** [comment]
Well the first issue I see is that your GPs and GLO messages are in fact empty.  [You are sending a valid message frame out at 1Hz but no SVs].  You are not in fact sending MT1230, Take a close look at your setup.   I found that when I displayed the messages in the SNIP RTCM 3 decoder/viewer, but the SV counts being set to zero in your image above should have tipped me off.  Where is the other nearby Base, perhaps it can be used as rover your base once the signals are all working.

**Comment by baxter86811** [comment]
Yes, I see that now. What could it be wrong with my setup? I downloaded configuration file from Ardusimple site, port baudrate is set to 115200 for UART1 and UART2, Protocol In and Out is "0+1+5 UBX+NMEA+RTCM3" for UART1 and USB, "5-RTCM3" for UART2. Rate is set on 200 ms (5Hz). When on U-Center, base module shows TIME on the interface.
BTW should I use MSM4 also or I should leave it just with MSM7 messages?

**Comment by baxter86811** [comment]
I steal can't get it to work right... I switched to MSM4 messages, and when I connect the board to Windows PC and start STRSVR everything looks fine (I now have the SVs posted, but still no 1230)
![image](https://user-images.githubusercontent.com/68125356/87244056-42bcb780-c43b-11ea-97b2-8418899833fa.png)

But, when I connect the board to RPi and start str2str I lose the SVs. I also have a problem when sending data for 1094 I get an error message: "segmentation fault" and it stops. When I type in -msg all the messages except 1094 str2str starts streaming but on the SNIP monitor I get no SVs

**Comment by baxter86811** [comment]
As you can see: str2str prints segmentation fault with 1094, but streams with 1097 (both set in U-center on UART1 and USB 1 sec)
![Capture](https://user-images.githubusercontent.com/68125356/87245038-46077180-c442-11ea-8e2c-aca7a7445b4b.PNG)
But no SVs:
![image](https://user-images.githubusercontent.com/68125356/87245055-6a634e00-c442-11ea-83f8-f9fbc380ff90.png)

**Comment by Montesaltos** [comment]
It looks like you are running into a problem caused by a bug in the latest released firmware of the ZED 9FP which corrupts the RTCM messages when you use different settings to output RTCM on different ports. 
As the base configuration delivered by Ardusimple sets 1084, 94, etc.. on port 2, any different settings of messages for the USB port will cause troubles.

this is even stated at the U-blox release note of the firmwares "If the receiver is configured to output RTCM messages on several ports, the ports must have the same RTCM configuration,  otherwise the MSM multiple message bit might not be set correctly"  https://www.u-blox.com/en/docs/UBX-20019211

Just try to set the exact same configuration on the ports using U-center to confirm.

**Comment by DavidKelleySCSC** [comment]
Perhaps what Montesaltos states is the key insight here for you.  Looking at your current data stream a moment ago, the content looks good and you have valid messages for GPS/GLO/GAL in MSM7 forms.  And plots out (at least the GPS part) as a coherent data set.  Looks like it is working to me. 

![image](https://user-images.githubusercontent.com/5143060/88556914-67d62c00-cfde-11ea-8bf7-74dce274ae79.png)

**Comment by baxter86811** [comment]
Hello, I'm in the middle of harvesting season so I didn't had time to write how I got around this. Yes, I'm guessing there's some bug in the firmware with the message outputting. As I wrote above, I tried to filter out RTCM3 messages in STR2STR, but whatever I did RTKLib pushed every outputted message from module to rtk2go (RTCM, NMEA, UBX) so my stream to rtk2go was corrupted and unusable. So I went to U-Center/Messages and disabled ALL messages (right click/Disable Child Messages so they got light gray) except RTCM messages I want to go out: 1005, 1077, 1087, 1097 and 1230. Then I started STR2STR without "-msg "1005(1),1077(1),1087(1),1097(1),1230(1)"" since my messages output it already filtered, and everything works as it should.
Later I had the same problem with my rover module when exporting NMEA messages to Cerea software. So I again disabled all messages and activated just GNRMC and GNGGA. Then I verified the stream in Packet Console
